rustls/src/common_state.rs

```diff
@@ -974,5 +979,40 @@ impl KxState {
     }
 }
 
+pub(crate) struct HandshakeFlight<'a, const TLS13: bool> {
```

Comment:
I briefly considered the restrictiveness of the const generic bool here vs something more expressive but decided it was premature optimization and this seems quite reasonable after all.